src/discordcr/mappings/guild.cr

```diff
@@ -237,6 +237,23 @@ module Discord
     end
   end
 
+  struct PartialRole
```

Comment:
I'm not sure this is a good name because every field from `Role` is here too but I think an additional type is required because here all the instance variable types are nilable, in `Role` they are not.
Something like `GuildCreationRole` might be too long.